rbac/management/group/definer.py

```diff
@@ -78,29 +79,39 @@ def seed_group():
         update_group_roles(admin_group, admin_roles, public_tenant)
         logger.info("Finished seeding default org admin group %s.", name)
 
+    return group, admin_group
 
-def set_system_flag_before_update(group, tenant, user):
+
+def set_system_flag_before_update(group: Group, tenant, user) -> Optional[Group]:
     """Update system flag on default groups."""
     if group.system:
-        group = clone_default_group_in_public_schema(group, tenant)
+        group = clone_default_group_in_public_schema(group, tenant)  # type: ignore
         group_flag_change_notification_handler(user, group)
     return group
 
 
-def clone_default_group_in_public_schema(group, tenant):
+def clone_default_group_in_public_schema(group, tenant) -> Optional[Group]:
     """Clone the default group for a tenant into the public schema."""
+    if settings.PRINCIPAL_CLEANUP_UPDATE_ENABLED_UMB:
+        # TODO: bootstrap the tenant to get the mapping
+        # use this for uuid instead and to remove the default role binding tuple
+        uuid = uuid4()
+    else:
+        uuid = uuid4()
+
     public_tenant = Tenant.objects.get(tenant_name="public")
     tenant_default_policy = group.policies.get(system=True)
     group.name = "Custom default access"
     group.system = False
     group.tenant = tenant
-    group.uuid = uuid4()
+    group.uuid = uuid
```

Comment:
Looks like this probably isn't an issue currently, but wondering if we might want to name it something more explicit to not get in a weird situation where we're stepping on the `uuid` module?

Reply:
yeah great point, simple fix